### PR TITLE
Remove duplicate MISSING_TABLE errorcode

### PR DESCRIPTION
**Pull request by jainxrohit**
Removing the MISSING_TABLE error code from StandardErrorCode, as it was already available in the SemanticErrorCode. Refactoring SemanticException so that it only stores NodeLocation instead of the Node class. It still keeps the constructor to take Node as a parameter to avoid too many changes.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/SemanticException.java [hunk 1]
```diff
@@ -16,39 +16,51 @@
...
 
     public SemanticException(SemanticErrorCode code, Node node, String format, Object... args)
     {
-        super(formatMessage(format, node, args));
+        this(code, node == null ? Optional.empty() : node.getLocation(), format, args);
+    }
+
+    public SemanticException(SemanticErrorCode code, Optional<NodeLocation> location, String format, Object... args)
+    {
+        super(formatMessage(format, location, args));
```
Comment (ajaygeorge) [inline presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/SemanticException.java:42]: is location guaranteed to be non-null here?
do we need a requireNonNull for `location`

**Review by ajaygeorge — APPROVED** [review]
LGTM % a few comments

File: presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/SemanticException.java [hunk 2]
```diff
@@ -16,39 +16,51 @@
...
+
+    public SemanticException(SemanticErrorCode code, String format, Object... args)
+    {
+        this(code, Optional.empty(), format, args);
+    }
 
     public SemanticException(SemanticErrorCode code, Node node, String format, Object... args)
     {
-        super(formatMessage(format, node, args));
+        this(code, node == null ? Optional.empty() : node.getLocation(), format, args);
```
[the remark this answers is not in the document]
Comment (jainxrohit) [reply presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/SemanticException.java:37]: Here we are checking null on node, however Optional is created for nodeLocation.

File: presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/SemanticException.java (hunk 1 above)
Comment (jainxrohit) [reply presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/SemanticException.java:42]: Location is not null in the node class.

**Review by highker — APPROVED** [review]

**Pull request merged**